chain/fil/builder/builder.go

```diff
@@ -0,0 +1,53 @@
+package builder
+
+import (
+	xc "github.com/cordialsys/crosschain"
+	xcbuilder "github.com/cordialsys/crosschain/builder"
+	tx "github.com/cordialsys/crosschain/chain/fil/tx"
+	filinput "github.com/cordialsys/crosschain/chain/fil/tx_input"
+)
+
+// TxBuilder for filecoin
+type TxBuilder struct {
+	Asset xc.ITask
+}
+
+type TxInput = filinput.TxInput
+
+var _ xcbuilder.FullTransferBuilder = TxBuilder{}
+
+// NewTxBuilder creates a new filecoin TxBuilder
+func NewTxBuilder(cfgI xc.ITask) (TxBuilder, error) {
+	return TxBuilder{
+		Asset: cfgI,
+	}, nil
+}
+
+// NewTransfer creates a new transfer
+func (txBuilder TxBuilder) Transfer(args xcbuilder.TransferArgs, input xc.TxInput) (xc.Tx, error) {
+	return txBuilder.NewTransfer(args.GetFrom(), args.GetTo(), args.GetAmount(), input)
+}
+
+// Old transfer interface
+func (txBuilder TxBuilder) NewTransfer(xcFrom xc.Address, xcTo xc.Address, amount xc.AmountBlockchain, input xc.TxInput) (xc.Tx, error) {
+	txInput := input.(*TxInput)
+	from := string(xcFrom)
+	to := string(xcTo)
+
+	message := tx.Message{
+		Version:    0,
+		To:         to,
+		From:       from,
+		Nonce:      txInput.Nonce,
+		Value:      xc.NewAmountBlockchainFromUint64(amount.Int().Uint64()),
+		GasLimit:   txInput.GasLimit,
+		GasFeeCap:  txInput.GasFeeCap,
+		GasPremium: txInput.GasPremium,
+		Method:     0,
+		Params:     []byte{},
```

Comment:
Perhaps use a constructor that takes `tx_input` + `xcbuilder.TransferArgs` as a input

```go
func NewMessage(args xcbuilder.TransferArgs,  input tx_input.TxInput)
```